src/ray/gcs/gcs_client/service_based_accessor.cc

```diff
@@ -916,46 +906,44 @@ Status ServiceBasedTaskInfoAccessor::AsyncSubscribeTaskLease(
     const TaskID &task_id,
     const SubscribeCallback<TaskID, boost::optional<rpc::TaskLeaseData>> &subscribe,
     const StatusCallback &done) {
-  RAY_LOG(DEBUG) << "Subscribing task lease, task id = " << task_id;
   RAY_CHECK(subscribe != nullptr)
       << "Failed to subscribe task lease, task id = " << task_id;
 
+  auto fetch_data_operation = [this, task_id, subscribe](const StatusCallback &done) {
+    auto callback = [task_id, subscribe, done](
+                        const Status &status,
+                        const boost::optional<rpc::TaskLeaseData> &result) {
+      subscribe(task_id, result);
+      if (done) {
```

Comment:
when calling `fetch_data_operation` during `resubscribe`, `done` shouldn't be called again, right?

Reply:
I have change done to fetch_done.